llvm/lib/Target/AArch64/AArch64FrameLowering.cpp

```diff
@@ -3072,54 +3094,65 @@ bool AArch64FrameLowering::spillCalleeSavedRegisters(
       std::swap(FrameIdxReg1, FrameIdxReg2);
     }
 
-    unsigned PairRegs;
-    unsigned PnReg;
     if (RPI.isPaired() && RPI.isScalable()) {
-      PairRegs = AArch64::Z0_Z1 + (RPI.Reg1 - AArch64::Z0);
+      unsigned PairRegs = AArch64::Z0_Z1 + (RPI.Reg1 - AArch64::Z0);
```

Comment:
Suggest moving assignment to `PairRegs` closer to the place of use. A variable does not seem necessary, too.